lib/ssl-test.rb

```diff
@@ -69,48 +69,60 @@ def self.matching_domains domains, hostname
   # https://stackoverflow.com/questions/16244084/how-to-programmatically-check-if-a-certificate-has-been-revoked#answer-16257470
   # Returns an array with [ocsp_check_failed, certificate_revoked, error_reason, revocation_date]
   def self.test_ocsp_revocation cert, chain, open_timeout: 5, read_timeout: 5, redirection_limit: 5
+    @ocsp_response_cache ||= {}
     chain[0..-2].each_with_index do |current_checked_cert, i|
+      # https://tools.ietf.org/html/rfc5280#section-4.1.2.2
+      # The serial number [...] MUST be unique for each certificate issued by a given CA (i.e., the issuer name and serial number identify a unique certificate)
+      unicity_key = [current_checked_cert.issuer.to_s, current_checked_cert.serial.to_s]
```

Comment:
I wasn't sure but had the feeling that constructing an array here as hash key would be less efficient than concatenating the strings so I checked and it's the case:
```ruby
require 'benchmark'

str1 = "yolo"
str2 = "yata"

N = 1_000_000

Benchmark.bm do |x|
  x.report("array") {
    h = {}
    N.times {
      key = [str1, str2]
      h[key] = {}
    }
  }
  x.report("string") {
    h = {}
    N.times {
      key = "#{str1}/#{str2}"
      h[key] = {}
    }
  }
end

# $ ruby bench.rb
#         user     system      total        real
#  array  0.714008   0.000000   0.714008 (  0.714285)
# string  0.180115   0.000000   0.180115 (  0.180154)
```
Not a big deal but better using a single string here